### PR TITLE
Add support for sonata-project/block-bundle 4.x

### PR DESCRIPTION
**Pull request by core23**
## Subject

I am targeting this branch, because this is BC.

## Changelog

```markdown
### Added
- Add support for `sonata-project/block-bundle` 4.x
```

### CODE REVIEW

File: src/Block/FormatterBlockService.php [hunk 1]
```diff
@@ -65,12 +77,12 @@ public function configureSettings(OptionsResolver $resolver): void
         ]);
     }
 
-    public function getBlockMetadata($code = null): Metadata
```
Comment (franmomu) [inline src/Block/FormatterBlockService.php:68]: I guess there should be an upgrade note with these BC breaks.

Comment (core23) [reply src/Block/FormatterBlockService.php:68]: Not sure if this is a BC break, because both block version will work

Comment (franmomu) [reply src/Block/FormatterBlockService.php:68]: I meant because of removing public methods.

Comment (core23) [reply src/Block/FormatterBlockService.php:68]: Reverted it

Comment (VincentLanglet) [reply src/Block/FormatterBlockService.php:68]: If it's deprecated, it should trigger a deprecation.

Comment (VincentLanglet) [reply src/Block/FormatterBlockService.php:68]: And what about a `NEXT_MAJOR` comment to remove it.

Comment (core23) [reply src/Block/FormatterBlockService.php:68]: > And what about a `NEXT_MAJOR` comment to remove it.

Adding a `NEXT_MAJOR` comment is redundant for a deprecated method

Comment (VincentLanglet) [reply src/Block/FormatterBlockService.php:68]: Decision in dev-kit was to use both https://github.com/sonata-project/dev-kit/issues/206

As an example, when I tried to remove some deprecation on SonataAdmin@master, I was only looking for `NEXT_MAJOR` in my IDE.

**Review by VincentLanglet — APPROVED** [review]

**Comment by jordisala1991** [comment]
Keep in mind there is a circular reference here, and you are not testing against block bundle 4

**Comment by VincentLanglet** [comment]
> Keep in mind there is a circular reference here, and you are not testing against block bundle 4

"sonata-project/admin-bundle": "^3.31" wont help indeed.

**Comment by timwentzell** [comment]
Hey guys - I just stumbled across this PR as I was attempting to upgrade my block bundle to v4.4. It looks as though only the PHP 8.0 tests failed - just curious if this is still a work in progress? Thanks!

**Comment by core23** [comment]
> Hey guys - I just stumbled across this PR as I was attempting to upgrade my block bundle to v4.4. It looks as though only the PHP 8.0 tests failed - just curious if this is still a work in progress? Thanks!

It's still WIP. I don't have much time at the moment. Feel free to continue working on this PR.

**Review by VincentLanglet — APPROVED** [review]

**Review by franmomu — APPROVED** [review]

**Comment by VincentLanglet** [comment]
Hi @core23, seems like you're in the team release-manager, so you have the rights to create a release
https://github.com/orgs/sonata-project/teams/release-managers

**Pull request merged**